addons/website_sale/static/src/js/website_sale.js

```diff
@@ -326,7 +326,7 @@ publicWidget.registry.WebsiteSale = publicWidget.Widget.extend(VariantMixin, car
         var attributeIds = _.map($attributes, function (elem) {
             return $(elem).data('value_id');
         });
-        history.replaceState(undefined, undefined, '#attr=' + attributeIds.join(','));
+        window.location.hash = 'attr=' + attributeIds.join(',');
```

Comment:
It is, as replacing history state cannot be listened from the client action to update the browser's url.
I did not find other occurrences of a history.replaceState than these two

Reply:
window.location.href will reload the window, so it will be caught in the 'load' event and the browser's url will be correctly updated (`hashchange` and `load` are the two events updating the browser's url)

Reply:
Ok, strange. So the previous code allowed to replace the hash without triggering an hashchange event? Weird.

> window.location.href will reload the window, so it will be caught in the 'load' event 

Unrelated but I just tested: if the href is an external one this is not detected and loaded inside the iframe. Maybe no usecase in our code (and I guess coders can just do `window.top.location.href = my_external_link` in that potential case to solve the issue)  :grimacing: But have we no way to solve that automatically ?

Reply:
Unfortunately, I don't think so... We can't really listen to redirections from the browser for security reasons,
And if the iframe fails the error won't be verbose enough to know where we can redirect the top window (we won't have the destination).

But maybe, instead of using window.location.href of the global window object, we could use the one from the "browser" odoo object `/web/static/src/core/browser/browser.js`. This way, we could patch it in the website root to redirect in the top window it is external and a frameElement

Reply:
> And if the iframe fails the error won't be verbose enough to know where we can redirect the top window

Well, if you let the iframe be loaded, you could just retrieve the destination with `iframe.src` or something similar, no? but that would be too late anyway.

> But maybe, instead of using window.location.href of the global window object, we could use the one from the "browser" odoo object /web/static/src/core/browser/browser.js. This way, we could patch it in the website root to redirect in the top window it is external and there is a frameElement

I don't understand what you mean, why could not you do just `window.top.location.href = ...` in that case?

Reply:
> After the iframe fails to load because of a cross-origin request, it will have this kind of url: chrome-error://chromewebdata/

Ah :open_mouth: 

> I was just thinking it could be an option to use the browser, a patchable object, rather than already knowing the implementation detail of the website and using window.top

Ok. In this case, using window.top makes sense from the start, it's not "knowing the implementation detail of website" IMO :wink: Anyways, we'll see if that problem ever presents itself. I am actually more worried about usecases where the redirection is done by library code or something like that. Something like "following the checkout process in the eCommerce, redirect to an external payment process => more than probably not allowed in an iframe anyway => crash... but I guess worst case scenario, we'll patch the page using the lib to display a warning saying this should be used outside of the Odoo iframe.

Is this PR ready (this was a side-debate)?

Reply:
Yes I don't really know how we could catch such redirections... if an odoo controller redirects to stripe as you were saying... 
I tried to rework the `sendRequest` util, to use the fetch function of the top window, and test the url returned from the response, see if it is external or not, but could not get it to work yet...

Yes, the PR is ready :)